WIP: Clarifying Processing engine instances (#5940)
influxdata/docs-v2: content/shared/v3-core-plugins/_index.md

  --object-store [OBJECT_STORE_TYPE] \
  --plugin-dir /path/to/plugins
```
Replace `/path/to/plugins` with the directory where you want to store your Python plugin files. All plugin files must be located in this directory or its subdirectories.

> ### Configure distributed environments

Reviewer: This mostly, though not strictly, applies to Enterprise. We may need to dupe this page and move them into core and enterprise product directories.

Reviewer: We should use one of the new `show-in` or `hide-in` shortcodes that Scott deployed yesterday.

If you're running multiple InfluxDB instances (distributed deployment):

1. Decide where plugins should run
- Data processing plugins run on ingestor nodes
- HTTP-triggered plugins run on nodes handling API requests
- Scheduled plugins can run on any configured node
2. Enable plugins on selected instances
3. Maintain identical plugin files across all instances where plugins run
 - Use shared storage or file synchronization tools to keep plugins consistent

> [Note] A single node can serve multiple roles (ingestor, querier, compactor), 

> > so configure the plugin directory based on where you want plugins to execute.

Reviewer: This seems questionable to me. Did this come from Pete?

Author: The information I used for this PR stems from a conversation I had with Pete. I slacked you the details about it last Friday. It could be my interpretation that is a little unclear. I can remove the last sentence? @jstirnaman

Reviewer: > The information I used for this PR stems from a conversation I had with Pete. I slacked you the details about it last Friday. It could be my interpretation that is a little unclear. I can remove the last sentence? @jstirnaman

OK, I see. Sorry I missed your message.

  --plugin-dir /path/to/plugins
```

Replace `/path/to/plugins` with the directory where you want to store your Python plugin files. All plugin files must be located in this directory or its subdirectories.

### Configure distributed environments

If you're running multiple InfluxDB instances (distributed deployment):

> 1. Decide where plugins should run
> - Data processing plugins run on ingestor nodes
> - HTTP-triggered plugins run on nodes handling API requests
> - Scheduled plugins can run on any configured node

Reviewer: ```suggestion
1. Decide where plugins should run:

    - Data processing plugins, such as WAL plugins, run on ingestor nodes
    - HTTP-triggered plugins run on nodes handling API requests
    - Scheduled plugins can run on any configured node
```

2. Enable plugins on selected instances

> 3. Maintain identical plugin files across all instances where plugins run
>  - Use shared storage or file synchronization tools to keep plugins consistent

Reviewer: ```suggestion
3. Maintain identical plugin files across all instances where plugins run. Use shared storage or file synchronization tools to keep plugins consistent
```

> > [Note] A single node can serve multiple roles (ingestor, querier, compactor), 
> > so configure the plugin directory based on where you want plugins to execute.

Reviewer: ```suggestion
> [!Note]
> #### Provide plugins to nodes that run them
>
> Configure your plugin directory on the same system as the nodes that run the triggers and plugins.
```

  --plugin-dir /path/to/plugins
```

Replace `/path/to/plugins` with the directory where you want to store your Python plugin files. All plugin files must be located in this directory or its subdirectories.

> {{% show-in "enterprise" %}}
> 
> ### Configure distributed environments
> 
> If you're running multiple InfluxDB instances (distributed deployment):
> 
> 1. Decide where plugins should run
>    - Data processing plugins, such as WAL plugins, run on ingestor nodes
>    - HTTP-triggered plugins run on nodes handling API requests
>    - Scheduled plugins can run on any configured node
> 2. Enable plugins on selected instances
> 3. Maintain identical plugin files across all instances where plugins run
>    - Use shared storage or file synchronization tools to keep plugins consistent
> 
> > [!Note]
> > #### Provide plugins to nodes that run them
> >
> > Configure your plugin directory on the same system as the nodes that run the triggers and plugins.
> 
> {{% /show-in %}}

Reviewer: ```suggestion

### Configure distributed environments

If you're running multiple InfluxDB instances (distributed deployment):

1. Decide where plugins should run
{{% show-in "enterprise" %}}
   - Data processing plugins, such as WAL plugins, run on ingestor nodes
   - HTTP-triggered plugins run on nodes handling API requests
   - Scheduled plugins can run on any configured node
{{% /show-in %}}
2. Enable plugins on selected instances
3. Maintain identical plugin files across all instances where plugins run
   - Use shared storage or file synchronization tools to keep plugins consistent

> [!Note]
> #### Provide plugins to nodes that run them
>
> Configure your plugin directory on the same system as the nodes that run the triggers and plugins.

```

I don't know if that actually works or if it breaks the formatting. If it breaks, just dupe the list--for example:

```
{{% show-in "enterprise" %}}
1. Decide where plugins should run

   - Data processing plugins, such as WAL plugins, run on ingestor nodes
   - HTTP-triggered plugins run on nodes handling API requests
   - Scheduled plugins can run on any configured node
2. Enable plugins on selected instances
3. Maintain identical plugin files across all instances where plugins run
   - Use shared storage or file synchronization tools to keep plugins consistent
{{% /show-in %}}
```

```
{{% show-in "core" %}}
1. Decide where plugins should run
2. Enable plugins on selected instances
3. Maintain identical plugin files across all instances where plugins run
   - Use shared storage or file synchronization tools to keep plugins consistent
{{% /show-in %}}
```

The 3 bullets under `1` are the only points that are strictly specific to Enterprise--although Core isn't designed to be multi-node,  you could run multiple nodes and keep track yourself which nodes you use for writing and which for querying.

```

This creates a Python virtual environment in your plugins directory with the specified packages installed.

### Connect Grafana to your InfluxDB instance

> When configuring Grafana to connect to InfluxDB 3 Enterprise:

Reviewer: ```suggestion
When configuring Grafana to connect to an InfluxDB 3 Enterprise instance:
```

- **URL**: Use a querier URL or any node that serves queries

> - **Node roles**: In distributed deployments, a single node can serve multiple roles (ingest, queries, compaction)

Reviewer: ```suggestion
```

Doesn't seem relevant here.

```

This creates a Python virtual environment in your plugins directory with the specified packages installed.

> ### Connect Grafana to your InfluxDB instance
> 
> When configuring Grafana to connect to an InfluxDB 3 Enterprise instance:
> 
> - **URL**: Use a querier URL or any node that serves queries
> 
> Example URL format: `https://querier.your-influxdb.com:8086`

Reviewer: ```suggestion
{{% show-in "enterprise" %}}
### Connect Grafana to your InfluxDB instance

When configuring Grafana to connect to an InfluxDB 3 Enterprise instance:

- **URL**: Use a querier URL or any node that serves queries

Example URL format: `https://querier.your-influxdb.com:8086`
{{% /show-in %}}
```
